x-pack/plugins/enterprise_search/public/applications/app_search/components/documents/search_experience/customization_modal.tsx

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState } from 'react';
+
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiComboBox,
+  EuiForm,
+  EuiFormRow,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiOverlayMask,
+  EuiText,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { useValues } from 'kea';
+
+import { EngineLogic } from '../../engine';
+
+interface Props {
+  filterFields: string[];
+  sortFields: string[];
+  onClose(): void;
+  onSave({ sortFields, filterFields }: { sortFields: string[]; filterFields: string[] }): void;
+}
+
+const fieldNameToComboBoxOption = (fieldName: string) => ({ label: fieldName });
+const comboBoxOptionToFieldName = ({ label }: { label: string }) => label;
+
+export const CustomizationModal: React.FC<Props> = ({
+  filterFields,
+  onClose,
+  onSave,
+  sortFields,
+}) => {
+  const { engine } = useValues(EngineLogic);
+
+  const [selectedFilterFields, setSelectedFilterFields] = useState(
+    filterFields.map(fieldNameToComboBoxOption)
+  );
+  const [selectedSortFields, setSelectedSortFields] = useState(
+    sortFields.map(fieldNameToComboBoxOption)
+  );
```

Comment:
Just at a glance, I'm somewhat surprised this works - I've typically had issues in the past with React components setting state from props. I'm guessing it's because the modal completely deletes itself / unmounts on save and completely re-mounts on open 🤔 Super interesting and somewhat clever!